File: Programs/databaseStuff.py
from WebsiteObject import Website
import sqlite3

connection = sqlite3.connect('C:\\Users\\Jonah\\Documents\\Third Year\\Big Project\\Databases\\webpages.db')
cursor = connection.cursor()

cursor.execute("SELECT * FROM webpages")

rows = cursor.fetchall()

# Process the data
for row in rows:
    print(row)

    # Turn each row into a new website
    newWebsite = Website(row[0], row[1], row[2], row[3], row[4], row[5])
    print(newWebsite.getTitle())

cursor.close()


cursor = connection.cursor()
newWebsite = Website("title", ["keyword1", "keyword2"], "url", "summary", 0, 0.0)
dataToWrite = []

for i in newWebsite.getKeywords():
    dataToWrite.append((newWebsite.getTitle(), i, newWebsite.getUrl(), newWebsite.getSummary(), 0, 0.0))

# Perform the insert query for each of the website's keywords
for i in range(len(dataToWrite)):
    cursor.execute("INSERT INTO webpages (title, keyword, URL, summary, relevanceScore, pageRank) VALUES (?, ?, ?, ?, ?, ?)", dataToWrite[i])
    connection.commit()

cursor.close()
connection.close()

File: Web Stuff/databaseStuff.php
<?php
error_reporting(E_ALL);
ini_set('display_errors', 'On');

include 'Website.php';

$db = new SQLite3('webpages.db');
if (!$db) {
    die("Failed to connect to the database.");
}

$result = $db->query('SELECT * FROM webpages');
if (!$result) {
    die("Query failed: " . $db->lastErrorMsg());
}

$websiteList = [];
$isFirstWebsite = true;
$currentKeywordList = [];
$previousWebsite = null;
$counter = 0;

while ($row = $result->fetchArray()) {
	
	$currentWebsite = $row['url'];
	
	if (!$isFirstWebsite) {
		if ($currentWebsite == $previousWebsite) {
			$currentKeywordList[] = $row['keyword'];
		} else {
			$newWebsite = new Website($row['title'], $currentKeywordList, $row['url'], $row['summary'], 0, 0.0);
			$websiteList[] = $newWebsite;
			$currentKeywordList = [];
			$currentKeywordList[] = $row['keyword'];
		}
	} else {
		$currentKeywordList[] = $row['keyword'];
		$isFirstWebsite = false;
	}
	
	$previousWebsite = $currentWebsite;

}

echo sizeOf($websiteList);

for ($i=0; $i<sizeOf($websiteList); $i++) {
  echo $websiteList[$i]->getTitle();
	echo  "<br>";
  for ($j=0; $j<sizeOf($websiteList[$i]->getKeywords()); $j++) {
	  echo $websiteList[$i]->getKeywords()[$j];
	  echo "<br>";
  }
  echo "<br>";
}

$db->close();
?>
